cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -278,6 +278,60 @@ struct elementwise_aggregator {
   }
 };
 
+/**
+ * @brief Function object to update a single element in a target column using
+ * the dictionary key addressed by the specific index.
+ *
+ * `target[target_index] = d_dictionary.keys[d_dictionary.indices[source_index]]`
+ */
+struct update_target_from_dictionary {
+  template <typename KeyType>
+  std::enable_if_t<is_fixed_width<KeyType>() && !is_fixed_point<KeyType>()> __device__
+  operator()(mutable_column_device_view& target,
+             size_type& target_index,
+             column_device_view& d_dictionary,
+             size_type& source_index) const noexcept
+  {
+    auto const keys  = d_dictionary.child(1);
+    auto const value = keys.element<KeyType>(
+      static_cast<cudf::size_type>(d_dictionary.element<dictionary32>(source_index)));
+    using Target = target_type_t<KeyType, aggregation::SUM>;
+    atomicAdd(&target.element<Target>(target_index), static_cast<Target>(value));
```

Comment:
Try to reuse the existing code with something like
```
auto const new_source_index = static_cast<cudf::size_type>(d_dictionary.element<dictionary32>(source_index));
update_target_element<KeyType, k, target_has_nulls, source_has_nulls>{}(target, target_index, keys, new_source_index);
```

Reply:
I could not get this to work. The only specialization is needed for `Source=dictionary32` and `k=SUM`. The rest of work fine since they either just count group values are find the indexes of the extrema. This means that these can just use the dictionary column's indices which works by default without any changes. The `SUM` agg needs the actual values from the child keys column so that specialization is the only one required.